### PR TITLE
Dockerfile: add dependencies for lede environment

### PR DESCRIPTION
**Pull request by nikhil-zinjurde-imgtec**
Include additional dependencies as mentioned in https://lede-project.org/docs/user-guide/imagebuilder#prerequisites

Signed-off-by: Nikhil Zinjurde <nikhil.zinjurde@imgtec.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,12 +6,12 @@ MAINTAINER IMGTEC
 # Add multiarch support
 RUN dpkg --add-architecture i386
 
-# Openwrt dependencies
+# Openwrt/LEDE dependencies
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
 binutils flex perl gawk grep file openssl bzip2 unzip zip rsync bc \
-build-essential bsdmainutils cmake device-tree-compiler \
+build-essential bsdmainutils ccache cmake gettext xsltproc wget \
```
Comment (Ham22) [inline Dockerfile]: I have left ccache disabled in the builds as it is unlikely to give us any benefit due to the current build system arch of using containers. We can enable and possibly mount the cache out to a volume to reap the full benefits.

File: Dockerfile [hunk 2]
```diff
@@ -6,12 +6,12 @@ MAINTAINER IMGTEC
 # Add multiarch support
 RUN dpkg --add-architecture i386
 
-# Openwrt dependencies
+# Openwrt/LEDE dependencies
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
 binutils flex perl gawk grep file openssl bzip2 unzip zip rsync bc \
-build-essential bsdmainutils cmake device-tree-compiler \
```
Comment (Ham22) [inline Dockerfile]: Why are you moving wget and device-tree-compiler, the lines are attempting to have a logical split of 
stuff that builds things and stuff that gets things which this change breaks needlessly.

**Review by Ham22 — CHANGES_REQUESTED** [review]
Where did this list of changes come from?
The official docs provide a table of known deps here https://lede-project.org/docs/guide-developer/install-buildsystem#table_of_known_prerequisites. Some seem to be missing.

File: Dockerfile (hunk 2 above)
Comment (nikhil-zinjurde-imgtec) [reply Dockerfile]: Restored the logical split

**Comment by Ham22** [comment]
Btw does this fix any known issues? Whats the motivation?

**Comment by nikhil-zinjurde-imgtec** [comment]
This is to have a build environment to build LEDE from source, SDK and ImageBuilder.  There are not known issues yet without including these dependencies

File: Dockerfile [hunk 3]
```diff
@@ -6,10 +6,11 @@ MAINTAINER IMGTEC
 # Add multiarch support
 RUN dpkg --add-architecture i386
 
-# Openwrt dependencies
+# Openwrt/LEDE dependencies
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
 binutils flex perl gawk grep file openssl bzip2 unzip zip rsync bc \
 build-essential bsdmainutils cmake device-tree-compiler \
+asciidoc fastjar gettext intltool patch ruby sdcc xsltproc \
```
Comment (Ham22) [inline Dockerfile]: Coud you drop:
* patch is part of build-essential
* sdcc is a constrained device compiler for devices we will never use

**Review by Ham22 — APPROVED** [review]

**Pull request merged**